### PR TITLE
feat: add support for textDocument/linkedEditingRange

### LINKED ISSUE #179: add support for linkedEditingRange for react tsx tags
Spec: https://microsoft.github.io/language-server-protocol/specification#textDocument_linkedEditingRange
Client Implementation: https://github.com/prabirshrestha/vim-lsp/pull/1022

Demo with a different language server.

https://user-images.githubusercontent.com/629908/103635937-993f2480-4f8c-11eb-873a-63b7f31986d5.mp4

### PR DESCRIPTION
**Pull request by predragnikolic**
Typescript 5.1 introduced [linked editing](https://devblogs.microsoft.com/typescript/announcing-typescript-5-1/#linked-cursors-for-jsx-tags).

----

I implemented the request handling,
but I am not able to test it.

Does anybody know a client that uses typescript-language-server and supports linked editing?

I tried to test this, but I was **only** able to test the response, 
with this [experimental branch](https://github.com/sublimelsp/LSP/tree/experimental/try-linked-editing)(it doesn't work if you try to type,you can just see the response) in Sublime Text LSP. 

-----

Apart from that, I bumped the typescript version to 5.1 in this repo.
Currently that broke running `yarn lint`, **@typescript-eslint/parser** and I am not able to fix it now, (see the commit description 8325fb95c3f4d2c868dbec8bd6f584f33646dbcf)

TODO; 
- Test this at least with one client.
- Fix Eslint.

Feel free to push changes to this branch.

closes #179

### CODE REVIEW

**Comment by rchl** [comment]
The eslint think should be just a warning and should affect linting.
BTW. I prefer to bump dependencies through dedicated PRs.

**Comment by rchl** [comment]
I've updated all dependencies now.

File: src/lsp-server.ts [hunk 1]
```diff
@@ -1308,6 +1312,24 @@ export class LspServer {
             params.textDocument.uri, params.range, this.documents, this.tspClient, this.options.lspClient, this.configurationManager, token);
     }
 
+    async linkedEditingRange(params: lsp.LinkedEditingRangeParams, token?: lsp.CancellationToken): Promise<lsp.LinkedEditingRanges> {
```
Comment (rchl) [inline src/lsp-server.ts:1315]: I see that the lsp libraries define the return value as `LinkedEditingRanges | null`. What happens if we return `null`?

**Comment by rchl** [comment]
A simple test would be good to have.

File: src/lsp-server.ts (hunk 1 above)
Comment (predragnikolic) [reply src/lsp-server.ts]: Yes, I missed that.
I pushed a commit to return `null` in cases when ts can't compute LinkedEditingRanges.

**Comment by rchl** [comment]
thanks

**Pull request merged**